### PR TITLE
samples/terminate: add a sample to demonstrate wasm_runtime_terminate

### PR DESCRIPTION
**Pull request by yamt**
This is basically a modified copy of the "shared-module" example.

The guts of terminate/wasm-apps/testapp.wat was from my private repo.
(via wasi-threads)

### CODE REVIEW

**Comment by yamt** [comment]
the ci failure looks like https://github.com/bytecodealliance/wasm-micro-runtime/issues/2776
```
;; This module is required to validate, regardless of whether it can be
;; linked. Overloading is not possible in wasm itself, but it is possible
;; in modules from which wasm can import.
Writing WAST module to 'C:\Users\RUNNER~1\AppData\Local\Temp\tmppk5px9eg.wast'
Compiling WASM to 'C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm'
Running: ./wabt/out/gcc/Release/wat2wasm.exe --enable-thread --no-check C:\Users\RUNNER~1\AppData\Local\Temp\tmppk5px9eg.wast -o C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm
Starting interpreter for module 'C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm'
Running: ../../../product-mini/platforms/windows/build/RelWithDebInfo/iwasm.exe --module-path=C:\Users\RUNNER~1\AppData\Local\Temp --heap-size=0 -v=0 --repl C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm
module is ....'(module
    (import "not wasm" "overloaded" (func))
    (import "not wasm" "overloaded" (func (param i32)))
    (import "not wasm" "overloaded" (func (param i32 i32)))
    (import "not wasm" "overloaded" (func (param i64)))
    (import "not wasm" "overloaded" (func (param f32)))
    (import "not wasm" "overloaded" (func (param f64)))
    (import "not wasm" "overloaded" (func (result i32)))
    (import "not wasm" "overloaded" (func (result i64)))
    (import "not wasm" "overloaded" (func (result f32)))
    (import "not wasm" "overloaded" (func (result f64)))
    (import "not wasm" "overloaded" (global i32))
    (import "not wasm" "overloaded" (global i64))
    (import "not wasm" "overloaded" (global f32))
    (import "not wasm" "overloaded" (global f64))
    (import "not wasm" "overloaded" (table 0 funcref))
    (import "not wasm" "overloaded" (memory 0))
  )'
exception is ....'unknown import'
Writing WAST module to 'C:\Users\RUNNER~1\AppData\Local\Temp\tmppk5px9eg.wast'
Compiling WASM to 'C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm'
Running: ./wabt/out/gcc/Release/wat2wasm.exe --enable-thread --no-check C:\Users\RUNNER~1\AppData\Local\Temp\tmppk5px9eg.wast -o C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm
Starting interpreter for module 'C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm'
Running: ../../../product-mini/platforms/windows/build/RelWithDebInfo/iwasm.exe --module-path=C:\Users\RUNNER~1\AppData\Local\Temp --heap-size=0 -v=0 --repl C:\Users\RUNNER~1\AppData\Local\Temp\tmpdrhkky4n.wasm
Removing tempfiles
failed with a non-zero return code 1

spec tests FAILED
Error: Process completed with exit code 1.
```

File: samples/terminate/src/main.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+static void *
+runner(void *vp)
+{
+    wasm_module_inst_t inst = vp;
+    bool ok = wasm_runtime_init_thread_env();
+    assert(ok);
+    wasm_application_execute_main(inst, 0, NULL);
+    return inst;
```
Comment (wenyongh) [inline samples/terminate/src/main.c:30]: Had better call `wasm_runtime_destroy_thread_env` before return?

Comment (yamt) [reply samples/terminate/src/main.c:31]: sure. i added it.

**Review by wenyongh — COMMENTED** [review]
LGTM

**Pull request merged**